### PR TITLE
Update StreamJsonRpc

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Review by JoeRobich — APPROVED** [review]

File: eng/Versions.props [hunk 1]
```diff
@@ -212,27 +212,27 @@
     <RoslynMicrosoftVisualStudioExtensionManagerVersion>0.0.4</RoslynMicrosoftVisualStudioExtensionManagerVersion>
     <SourceBrowserVersion>1.0.21</SourceBrowserVersion>
     <SystemBuffersVersion>4.5.1</SystemBuffersVersion>
-    <SystemCompositionVersion>1.0.31</SystemCompositionVersion>
+    <SystemCompositionVersion>6.0.0</SystemCompositionVersion>
```
Comment (CyrusNajmabadi) [inline eng/Versions.props:215]: that's quite the jump.  any concerns here? :)

File: src/VisualStudio/Setup.Dependencies/Roslyn.VisualStudio.Setup.Dependencies.csproj [hunk 2]
```diff
@@ -35,6 +35,8 @@
     <NuGetPackageToIncludeInVsix Include="System.Runtime.CompilerServices.Unsafe" PkgDefEntry="BindingRedirect" />
     <NuGetPackageToIncludeInVsix Include="System.Text.Encoding.CodePages" PkgDefEntry="BindingRedirect" />
     <NuGetPackageToIncludeInVsix Include="System.Threading.Tasks.Extensions" PkgDefEntry="BindingRedirect" />
+    <NuGetPackageToIncludeInVsix Include="Nerdbank.Streams" PkgDefEntry="BindingRedirect" />
+    <NuGetPackageToIncludeInVsix Include="StreamJsonRpc" PkgDefEntry="BindingRedirect" />
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Setup.Dependencies/Roslyn.VisualStudio.Setup.Dependencies.csproj:39]: can you explain this part to me ? (as someone woefully confused by this tech).

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/VisualStudio/Setup.Dependencies/Roslyn.VisualStudio.Setup.Dependencies.csproj (hunk 2 above)
Comment (tmat) [reply src/VisualStudio/Setup.Dependencies/Roslyn.VisualStudio.Setup.Dependencies.csproj:39]: This adds binding redirects when deploy the Dependencies VSIX with F5.

File: eng/Versions.props (hunk 1 above)
Comment (tmat) [reply eng/Versions.props:215]: VS repo uses 6.0.0 as well.

Comment (tmat) [reply eng/Versions.props]: So apparently there is a concern: https://github.com/dotnet/roslyn/issues/59128 :(

**Review by jasonmalinowski — APPROVED** [review]

**Pull request merged**